inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.hpp

```diff
@@ -24,7 +24,8 @@ std::pair<bool, std::string> compare_functions(
     const bool compareConstValues = false,
     const bool compareNames = false,
     const bool compareRuntimeKeys = false,
-    const bool comparePrecisions = true);
+    const bool comparePrecisions = true,
+    const bool compareAttribures = false);
```

Comment:
Why do you disable attributes comparison by default?

Reply:
Because we have some failed test now:
```
[3151/3151] TransformationTests.ConvertPrecision_Equal (722 ms)
FAILED TESTS (62/3151):
    1259 ms: ./ieFuncTests smoke_LPT/ConcatWithIntermediateTransformation.CompareFunctions/f32_Shape{1, 3, 9, 9}_asymmetric_u8_i8_UpdateLevel_notMultiChannels___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 2.55 }___256Shape{}_{ 1.275 }_{ 2.55 }_{ 1.275 }_{ 2.55 }____256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___256Shape{}_{ 1.275 }_{ 2.55 }_{ 128 }_{ 255 }___f32_{  }_Shape{}_undefined_{ 0.01 }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.01 }_Shape{}_undefined_
    1218 ms: ./ieFuncTests TransformationTests.ConvertTopK3I64Output1
    1307 ms: ./ieFuncTests smoke_LPT/MatMulTransformation.CompareFunctions/f32_Shape{4, 16, 384, 64}_Shape{4, 16, 64, 384}__1_1___f32_{ 127 }_Shape{}_undefined_{ 0.02 }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.03 }_Shape{}_undefined__i8___{ 127 }_Shape{}_f32_{  }_Shape{}_undefined_i8___{  }_Shape{}_undefined_{  }_Shape{}_undefined_f32_f32___{  }_Shape{}_undefined_{ 0.0006 }_Shape{}_undefined
    1305 ms: ./ieFuncTests TransformationTests.LowLatencyRNN
    1275 ms: ./ieFuncTests ReduceToReshapePoolReshape/ConvertReduceToPoolingTests.CompareFunctions/2
    1360 ms: ./ieFuncTests smoke_LPT/ClampTransformation.CompareFunctions/asymmetric_u8_i8_UpdateLevel_Shape{1, 3, 224, 224}_u8__f32_{ 128 }_Shape{}_undefined_{ 3 }_Shape{}_undefined___{  }_Shape{}_undefined_{  }_Shape{}_undefined
    1325 ms: ./ieFuncTests TransformationTests.GRUSequenceConversionTest
    1337 ms: ./ieFuncTests TransformationTests.RNNSequenceConversionTest
    1334 ms: ./ieFuncTests smoke_LPT/ConcatWithIntermediateTransformation.CompareFunctions/f32_Shape{1, 3, 9, 9}_asymmetric_u8_i8_UpdateLevel_notMultiChannels___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 2.55 }___256Shape{}_{ 0 }_{ 1.275 }_{ 0 }_{ 1.275 }____256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___256Shape{}_{ 0 }_{ 1.275 }_{ 0 }_{ 128 }___f32_{  }_Shape{}_undefined_{ 0.01 }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.01 }_Shape{}_undefined_
    1379 ms: ./ieFuncTests TransformationTests.ConvertMatMulTest5
    1132 ms: ./ieFuncTests ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/3
    1258 ms: ./ieFuncTests TransformationTests.ConvertMatMulTest7
    1345 ms: ./ieFuncTests TransformationTests.FullyConnectedBiasFusionTest2D
    1353 ms: ./ieFuncTests ReduceToReshapePoolReshape/ConvertReduceToPoolingTests.CompareFunctions/1
    1164 ms: ./ieFuncTests smoke_LPT/MatMulTransformation.CompareFunctions/f32_Shape{1, 16, 384, 64}_Shape{1, 16, 64, 384}__1_1___f32_{ 127.5 }_Shape{}_undefined_{ 0.02 }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.03 }_Shape{}_undefined__u8___{ 128 }_Shape{}_f32_{  }_Shape{}_undefined_i8___{  }_Shape{}_undefined_{  }_Shape{}_undefined_f32_f32___{  }_Shape{}_undefined_{ 0.0006 }_Shape{}_undefined
    1329 ms: ./ieFuncTests TransformationTests.NormalizeL2FusionWithAdd
    1327 ms: ./ieFuncTests smoke_LPT/ClampTransformation.CompareFunctions/asymmetric_u8_i8_UpdateLevel_Shape{1, 3, 224, 224}_u8__f32_{  }_Shape{}_undefined_{ 3 }_Shape{}_undefined___{  }_Shape{}_undefined_{  }_Shape{}_undefined
    1323 ms: ./ieFuncTests TransformationTests.LogSoftmaxDecomposition
    1340 ms: ./ieFuncTests ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0
    1305 ms: ./ieFuncTests smoke_LPT/ClampTransformation.CompareFunctions/asymmetric_i8_i8_UpdateLevel_Shape{1, 3, 224, 224}_i8__f32_{  }_Shape{}_undefined_{ 3 }_Shape{}_undefined___{  }_Shape{}_undefined_{  }_Shape{}_undefined
    1326 ms: ./ieFuncTests smoke_LPT/ClampTransformation.CompareFunctions/asymmetric_u8_i8_UpdateLevel_Shape{1, 3, 224, 224}_u8__f32_{ 128, 128, 128 }_Shape{}_undefined_{ 3, 3, 3 }_Shape{}_undefined___{  }_Shape{}_undefined_{  }_Shape{}_undefined
    1329 ms: ./ieFuncTests smoke_LPT/ConcatWithIntermediateWithConstantTransformation.CompareFunctions/asymmetric_u8_i8_UpdateLevel_Shape{1, 3, 9, 9}_notMultiChannels___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 2.55 }___256Shape{}_{ 1.275 }_{ 2.55 }_{ 1.275 }_{ 2.55 }____256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___256Shape{}_{ 1.275 }_{ 2.55 }_{ 128 }_{ 255 }__u8___{  }_Shape{}_undefined_{  }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.01 }_Shape{}_undefined_
    1198 ms: ./ieFuncTests TransformationTests.MulAddAddMulFusion
    1332 ms: ./ieFuncTests smoke_LPT/FakeQuantizeAndTwoOutputBranchesWithConvolutionTransformation.CompareFunctions/f32_Shape{1, 32, 72, 48}_asymmetric_u8_i8_UpdateLevel__256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 2.55 }____{  }_Shape{}_undefined_{ 1 }_Shape{1, 1, 1}_f32___{  }_Shape{}_undefined_{ 1 }_Shape{1, 1, 1}_f32
    1320 ms: ./ieFuncTests ReduceToReshapePoolReshape/ConvertReduceToPoolingTests.CompareFunctions/0
    1275 ms: ./ieFuncTests smoke_LPT/ConcatWithNeighborsTransformation.CompareFunctions/f32_Shape{1, 3, 9, 9}_asymmetric_u8_i8_UpdateLevel_multiChannels___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 2.55 }___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 1.275 }___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 0.85 }____256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___256Shape{}_{ 0 }_{ 2.55 }_{ 0 }_{ 255 }___f32_{  }_Shape{}_undefined_{ 0.01, 0.01, 0.01, 0.005, 0.005, 0.005 }_Shape{}_undefined__f32_{  }_Shape{}_undefined_{ 0.005, 0.005, 0.005, 0.00333, 0.00333, 0.00333 }_Shape{}_undefined_
    
```

```
Note: Google Test filter = ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ReduceToReshape/ConvertReduceToPoolingTests
[ RUN      ] ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0

MEM_USAGE=414412KB
../inference-engine/tests/functional/inference_engine/transformations/convert_reduce_to_pooling_test.cpp:119: Failure
Value of: res.first
  Actual: false
Expected: true
Comparison of attributes failed for nodes ReduceMax_72, Reshape_78 [cmp status: mismatch in value: special_zero]
[  FAILED  ] ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0, where GetParam() = (16-byte object <70-73 FF-67 19-56 00-00 60-73 FF-67 19-56 00-00>, { 2, 3, 64, 1 }, { 3 }, false, 72-byte object <20-78 FF-67 19-56 00-00 38-78 FF-67 19-56 00-00 38-78 FF-67 19-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00>) (3 ms)
[----------] 1 test from ReduceToReshape/ConvertReduceToPoolingTests (3 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (3 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0, where GetParam() = (16-byte object <70-73 FF-67 19-56 00-00 60-73 FF-67 19-56 00-00>, { 2, 3, 64, 1 }, { 3 }, false, 72-byte object <20-78 FF-67 19-56 00-00 38-78 FF-67 19-56 00-00 38-78 FF-67 19-56 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00>)

 1 FAILED TEST
[55/62] ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/0 returned/aborted with exit code 1 (1341 ms)
[56/62] ReduceToReshape/ConvertReduceToPoolingTests.CompareFunctions/1 (1352 ms)
Running main() from gtest_main.cc
```

Reply:
FYI `ieFuncTest` report:

Comparison of attributes failed for nodes Constant_82, Constant_86 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_89, Constant_98 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes MVN_69, MVN_76 [cmp status: mismatch in value: eps_mode : [OUTSIDE_SQRT] vs [INSIDE_SQRT]]
Comparison of attributes failed for nodes Interpolate_70, Interpolate_84 [cmp status: mismatch in value: coordinate_transformation_mode : [asymmetric] vs [align_corners]]
Comparison of attributes failed for nodes Constant_73, Constant_104 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_73, Constant_104 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes MVN_69, MVN_76 [cmp status: mismatch in value: eps_mode : [OUTSIDE_SQRT] vs [INSIDE_SQRT]]
Comparison of attributes failed for nodes MVN_69, MVN_76 [cmp status: mismatch in value: eps_mode : [OUTSIDE_SQRT] vs [INSIDE_SQRT]]
Comparison of attributes failed for nodes Constant_101, Constant_108 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_89, Constant_93 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Clamp_172, Clamp_160 [cmp status: mismatch in value: min : [0.000000] vs [128.000000]]
Comparison of attributes failed for nodes Reshape_81, Constant_86 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Parameter_68, Parameter_80 [cmp status: mismatch in value: element_type : [f32] vs [f64]]
Comparison of attributes failed for nodes topk.1, Convert_78 [cmp status: mismatch in value: destination_type : [i64] vs [i32]]
Comparison of attributes failed for nodes fakeQuantize2, FakeQuantize_116 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes Constant_95, Constant_98 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes fakeQuantize2, FakeQuantize_116 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes fakeQuantize2, FakeQuantize_116 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes TensorIterator_85/1/TensorIterator_85/Parameter_71/variable_1, ReadValue_114 [cmp status: mismatch in value: variable_id : [TensorIterator_85/Parameter_71/variable_1] vs [GRUTensorIterator/variable0]]
Comparison of attributes failed for nodes TensorIterator_85/1/TensorIterator_85/Parameter_71/variable_1, ReadValue_114 [cmp status: mismatch in value: variable_id : [TensorIterator_85/Parameter_71/variable_1] vs [RNNTensorIterator/variable0]]
Comparison of attributes failed for nodes Clamp_177, Clamp_165 [cmp status: mismatch in value: min : [0.000000] vs [128.000000]]
Comparison of attributes failed for nodes Divide_76, NormalizeL2_91 [cmp status: mismatch in value: eps : [1.000000] vs [0.000099]]
Comparison of attributes failed for nodes LSTMTensorIterator/1/LSTMTensorIterator/Parameter_72/variable_2, ReadValue_125 [cmp status: mismatch in value: variable_id : [LSTMTensorIterator/Parameter_72/variable_2] vs [LSTMTensorIterator/variable0]]
Comparison of attributes failed for nodes TensorIterator_88/1/TensorIterator_88/Parameter_72/variable_2, ReadValue_128 [cmp status: mismatch in value: variable_id : [TensorIterator_88/Parameter_72/variable_2] vs [LSTMTensorIterator/variable0]]
Comparison of attributes failed for nodes Clamp_168, Clamp_156 [cmp status: mismatch in value: max : [10.000000] vs [3.333333]]
Comparison of attributes failed for nodes Clamp_177, Clamp_165 [cmp status: mismatch in value: min : [0.000000] vs [128.000000]]
Comparison of attributes failed for nodes Clamp_168, Clamp_156 [cmp status: mismatch in value: max : [10.000000] vs [3.333333]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes Clamp_172, Clamp_160 [cmp status: mismatch in value: min : [0.000000] vs [128.000000]]
Comparison of attributes failed for nodes fakeQuantize2, FakeQuantize_116 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes Clamp_172, Clamp_160 [cmp status: mismatch in value: min : [0.000000] vs [128.000000]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes ReduceMax_72, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes Constant_204, Constant_168 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [128]]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes Constant_220, Constant_191 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes ReduceMax_72, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes Constant_125, Constant_76 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_220, Constant_191 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes fakeQuantize2, fakeQuantize2 [cmp status: mismatch in value: levels : [256] vs [129]]
Comparison of attributes failed for nodes ReduceMax_72/reshape_begin, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes Constant_192, Constant_161 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes ReduceMax_72/reshape_begin, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes Constant_78, Constant_74 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes ReduceMax_72, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes Constant_139, Constant_132 [cmp status: mismatch in value: value : look in to the mem buffer]
Comparison of attributes failed for nodes ReduceMax_72, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]
Comparison of attributes failed for nodes ReduceMax_72/reshape_begin, Reshape_78 [cmp status: mismatch in value: special_zero : [1] vs [0]]

Reply:
That's is really useful. I think we need to create a ticket to fix this tests.